src/rat/data_processing/newdata.py

```diff
@@ -55,9 +55,12 @@ def _determine_precip_link_and_version(date):
     version = _determine_precip_version(date)
     if version == "IMERG-FINAL":
         link = f"ftp://arthurhou.pps.eosdis.nasa.gov/gpmdata/{date.strftime('%Y')}/{date.strftime('%m')}/{date.strftime('%d')}/gis/3B-DAY-GIS.MS.MRG.3IMERG.{date.strftime('%Y%m%d')}-S000000-E235959.0000.V06A.tif"
-    elif version == "IMERG-LATE":
-        if date >= datetime(2022, 5, 8):  # Version was changed from V06B to V06C
-            link = f"https://jsimpsonhttps.pps.eosdis.nasa.gov/imerg/gis/{date.strftime('%Y')}/{date.strftime('%m')}/3B-HHR-L.MS.MRG.3IMERG.{date.strftime('%Y%m%d')}-S233000-E235959.1410.V06C.1day.tif"
+    elif version == "IMERG-LATE":        
+        if date >= datetime(2023, 7, 2):  # Version was changed from V06B to V06C
+            link = f"https://jsimpsonhttps.pps.eosdis.nasa.gov/imerg/gis/{date.strftime('%Y')}/{date.strftime('%m')}/3B-HHR-L.MS.MRG.3IMERG.{date.strftime('%Y%m%d')}-S233000-E235959.1410.V06D.1day.tif"
+        elif date >= datetime(2022, 5, 8):
+            link = f"https://jsimpsonhttps.pps.eosdis.nasa.gov/imerg/gis/{date.strftime('%Y')}/{date.strftime('%m')}/3B-HHR-L.MS.MRG.3IMERG.{date.strftime('%Y%m%d')}-S233000-E235959.1410.V06C.1day.tif"      
+        
```

Comment:
Add above line 67, the link change for dates before 1 July 2023 (basically the same thing you did for IMERG Late). Also, put comment on both the additions (IMERG early and IMERG late) : "Version was changed from V06C to V06D".